subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/DependencyInsightReportTask.java

```diff
@@ -393,63 +474,112 @@ private AttributeContainer concat(AttributeContainer configAttributes, Attribute
+                    ""
+                ));
+                if (!selected) {
+                    text.add("Compatible");
                 }
+                rows.add(new TablePrinter.Row(text, Info));
+            }
+            for (Map.Entry<Attribute<?>, AttributeMatchDetails> entry : bothAttributes.entrySet()) {
+                Object providedValue = attributes.getAttribute(entry.getKey());
+                AttributeMatchDetails match = entry.getValue();
+                List<String> text = new ArrayList<>(ImmutableList.of(
+                    entry.getKey().getName(),
+                    providedValue == null ? "" : providedValue.toString(),
+                    String.valueOf(entry.getValue().requestedValue)
+                ));
+                StyledTextOutput.Style style;
                 switch (match.matchType) {
-                    case requested:
+                    case REQUESTED:
+                        style = Success;
                         break;
-                    case different_value:
-                        out.withStyle(Info).text(" (compatible with: " + match.requestedValue + ")");
+                    case DIFFERENT_VALUE:
+                    case NOT_REQUESTED:
+                        style = Info;
                         break;
-                    case not_requested:
-                        out.withStyle(Info).text(" (not requested)");
+                    case INCOMPATIBLE:
+                        style = StyledTextOutput.Style.Error;
                         break;
+                    default:
+                        throw new IllegalStateException("Unknown match type: " + match.matchType);
                 }
-                out.println();
+                if (!selected) {
+                    text.add(match.matchType == MatchType.INCOMPATIBLE ? "Incompatible" : "Compatible");
+                }
+                rows.add(new TablePrinter.Row(text, style));
+            }
+            for (Attribute<?> attribute : requestedAttributes) {
+                Object requestedValue = requested.getAttribute(attribute);
+                List<String> text = new ArrayList<>(ImmutableList.of(
+                    attribute.getName(),
+                    "",
+                    String.valueOf(requestedValue)
+                ));
+                if (!selected) {
+                    text.add("Compatible");
+                }
+                rows.add(new TablePrinter.Row(text, Info));
             }
+
+            new TablePrinter(Strings.repeat(" ", 4), header, rows).print(out);
         }
 
-        private int computeAttributePadding(AttributeContainer attributes, AttributeContainer requested) {
-            int maxAttributeLen = 0;
-            for (Attribute<?> attribute : requested.keySet()) {
-                maxAttributeLen = Math.max(maxAttributeLen, attribute.getName().length());
-            }
+        private void writeFoundAttributes(StyledTextOutput out, AttributeContainer attributes) {
```

Comment:
You know, I'm wondering why anyone would want to run without the `--reasons` flag now.  The new table format contains all the original data, and the old format isn't being preserved exactly any anyways.

I know it's late in the game to bring this up, but, have we considered not making this an option and just using this format all the time?

Reply:
cc @big-guy, I think this might be a good idea since the table is now inline where the attributes used to be displayed.

Reply:
I think this is a good idea too.  One less thing that has to be remembered.